numpy/_core/tests/test_numeric.py

```diff
@@ -3200,6 +3200,57 @@ def test_timedelta(self):
         assert np.allclose(a, a, atol=0, equal_nan=True)
         assert np.allclose(a, a, atol=np.timedelta64(1, "ns"), equal_nan=True)
 
+    def test_tol_warnings(self):
+        a = np.array([1, 2, 3])
+        #Testing when rtol is infinite
+        with warnings.catch_warnings(record=True) as w:
+
+            warnings.simplefilter("always")
+            b = np.isclose(a, a, rtol=np.inf)
+            assert len(w) == 1
+            assert issubclass(w[-1].category, UserWarning)
+            assert "At least one of rtol and atol are infinite" in str(w[-1].message)
+        #Testing when atol is infinite
+        with warnings.catch_warnings(record=True) as w:
+
+            warnings.simplefilter("always")
+            b = np.isclose(a, a, atol=np.inf)
+            assert len(w) == 1
+            assert issubclass(w[-1].category, UserWarning)
+            assert "At least one of rtol and atol are infinite" in str(w[-1].message)
+        #Testing when both are infinite
+        with warnings.catch_warnings(record=True) as w:
+
+            warnings.simplefilter("always")
+            b = np.isclose(a, a, atol=np.inf, rtol=np.inf)
+            assert len(w) == 1
+            assert issubclass(w[-1].category, UserWarning)
+            assert "At least one of rtol and atol are infinite" in str(w[-1].message)
+        #Testing when atol is nan
+        with warnings.catch_warnings(record=True) as w:
+
+            warnings.simplefilter("always")
+            b = np.isclose(a, a, atol=np.nan)
+            assert len(w) == 1
+            assert issubclass(w[-1].category, UserWarning)
+            assert "At least one of rtol and atol are not a number" in str(w[-1].message)
+        #Testing when rtol is nan
+        with warnings.catch_warnings(record=True) as w:
+
+            warnings.simplefilter("always")
+            b = np.isclose(a, a, rtol=np.nan)
+            assert len(w) == 1
+            assert issubclass(w[-1].category, UserWarning)
+            assert "At least one of rtol and atol are not a number" in str(w[-1].message)
+        #Testing when bath are nan
+        with warnings.catch_warnings(record=True) as w:
+
+            warnings.simplefilter("always")
+            b = np.isclose(a, a, atol=np.nan,)
+            assert len(w) == 1
+            assert issubclass(w[-1].category, UserWarning)
+            assert "At least one of rtol and atol are not a number" in str(w[-1].message)
+
```

Comment:
There’s a lot of copy/paste here. Consider using a for loop or pytest parameterization so reviewers don’t have to carefully make sure all the repeated code is bug-free.